cypress/e2e/GUI/gui.cy.js

```diff
@@ -0,0 +1,141 @@
+/// <reference types="cypress" />
+describe('GUI Tests', () => {
+    beforeEach(function () {
+        cy.visit('/')
+        cy.contains('button', 'Accept').click()
+    })
+
+    it('Mostra as colunas Nome da Empresa e Acao, e oculta as colunas ID, Industria, Numero de Funcionarios e Tamanho em um viewport movel', () => {
+        cy.viewport(320, 480)
+        cy.contains('Company name').should('be.visible')
+        cy.contains('Action').should('be.visible')
+        cy.contains('Industry').should('not.be.visible')
+        cy.contains('Number of employees').should('not.be.visible')
+        cy.contains('Size').should('not.be.visible')
+    })
+
+    it('Exibe o cabecalho com um titulo, alternador de tema e um campo de entrada de texto', () => {
+        cy.contains('EngageSphere').should('be.visible')
+        cy.get('[class^="ThemeToggle_container"]').should('be.visible')
+        cy.get('input[id="name"]').should('be.visible')
+    })
+
+    it('Exibe a saudacao "Hi there" quando nenhum nome e fornecido', () => { 
+        cy.contains('h2', 'Hi there!').should('be.visible')
+    })
+
+    it('Exibe a saudacao "Hi Joe" quando o nome e fornecido', () => {
+        cy.get('input[data-testid="name"]').type('Joe')
```

Comment:
Mesmo que o comentário anterior.

```suggestion
        cy.get('#name').type('Joe')
```